File: src/main/java/io/github/bot/terminal/application/domains/common/conditions/factory/ConditionsFactory.java
package io.github.bot.terminal.application.domains.common.conditions.factory;


import io.github.bot.terminal.application.domains.common.conditions.entity.Condition;
import io.github.bot.terminal.application.domains.common.conditions.entity.ConditionDetails;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ConditionsFactory {



    public Condition build(ConditionDetails details) {
        return new Condition()
                .setConditionType(details.getConditionType())
                .setExpectedValue(details.getExpectedValue());
    }


}

File: src/main/kotlin/io/github/bot/terminal/application/domains/common/conditions/factory/ConditionsFactory.kt
package io.github.bot.terminal.application.domains.common.conditions.factory

import io.github.bot.terminal.application.domains.common.conditions.entity.Condition
import io.github.bot.terminal.application.domains.common.conditions.entity.ConditionDetails
import org.springframework.stereotype.Service

@Service
class ConditionsFactory {
    fun build(details: ConditionDetails): Condition =
            Condition(
                    conditionType = details.conditionType,
                    expectedValue = details.expectedValue
            )
}